frontend/packages/dev-console/src/components/pipelines/pipeline-builder/PipelineBuilderForm.tsx

```diff
@@ -149,9 +158,17 @@ const PipelineBuilderForm: React.FC<PipelineBuilderFormProps> = (props) => {
               </Button>
             </ActionGroup>
           </ButtonBar>
-          <Sidebar open={!!selectedTask} onRequestClose={() => setSelectedTask(null)}>
+          <Sidebar
+            open={!!selectedTask}
+            onRequestClose={() => {
+              if (selectedTask?.taskIndex === selectedTaskRef.current?.taskIndex) {
+                setSelectedTask(null);
+              }
+            }}
+          >
             {() => (
               <TaskSidebar
+                key={selectedTask.taskIndex}
```

Comment:
A comment would be good here to ensure in future someone doesn't remove the key because you actually want it remount.